crates/edr_napi/src/spec.rs

```diff
@@ -0,0 +1,139 @@
+use edr_eth::{
+    chain_spec::L1ChainSpec,
+    result::InvalidTransaction,
+    transaction::{IsEip155, IsEip4844, TransactionMut, TransactionType, TransactionValidation},
+};
+use edr_provider::{time::CurrentTime, ProviderError, ResponseWithTraces, SyncProviderSpec};
+use edr_rpc_client::jsonrpc;
+use napi::{Either, Status};
+use napi_derive::napi;
+
+use crate::trace::RawTrace;
+
+#[napi]
+pub struct Response {
+    // N-API is known to be slow when marshalling `serde_json::Value`s, so we try to return a
+    // `String`. If the object is too large to be represented as a `String`, we return a `Buffer`
+    // instead.
+    data: Either<String, serde_json::Value>,
+    /// When a transaction fails to execute, the provider returns a trace of the
+    /// transaction.
+    solidity_trace: Option<RawTrace>,
+    /// This may contain zero or more traces, depending on the (batch) request
+    traces: Vec<RawTrace>,
```

Comment:
It'd be good to have notes here that traces are only returned for L1.

Reply:
Addressed in https://github.com/NomicFoundation/edr/pull/633/commits/00209047e80270572b0e84be664fd175a14d8b65